### ISSUE TITLE
site controller on edge cluster crashes after upgrade to 0.5.1

### ISSUE THREAD

**Issue by kungfuchicken** [skupperproject/skupper#466]
After upgrading from 0.4.2  to 0.5.1 in both clusters (and creating new connection tokens and adding `cluster-local: "false"` to the site configmaps) skupper-site-controller goes into a crashloop fpr cluster configured as `edge: "true"` with this in the logs:

```
2021/03/27 21:09:44 Skupper site controller watching all namespaces
2021/03/27 21:09:44 Starting the Skupper site controller informers
2021/03/27 21:09:44 Waiting for informer caches to sync
2021/03/27 21:09:44 Checking if sites need updates (0.5.1)
2021/03/27 21:09:44 Version update not required for namespace "monitoring"
2021/03/27 21:09:44 Version update not required for namespace "default"
2021/03/27 21:09:44 Starting workers
2021/03/27 21:09:44 Started workers
2021/03/27 21:09:45 Skupper site exists monitoring/skupper-site
2021/03/27 21:09:45 Checking tokens...
2021/03/27 21:09:45 Checking token requests...
2021/03/27 21:09:45 Done.
2021/03/27 21:09:47 Skupper site exists default/skupper-site
2021/03/27 21:09:48 Checking tokens...
2021/03/27 21:09:48 Checking token requests...
2021/03/27 21:09:48 Done.
2021/03/27 21:09:48 Connecting site in monitoring using token monitoring-connection-token
E0327 21:09:48.557352       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 88 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13223a0, 0x200fe10)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x13223a0, 0x200fe10)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/skupperproject/skupper/client.(*VanClient).ConnectorCreate.func1(0x0, 0x0)
        /go/src/app/client/connector_create.go:196 +0x22f
k8s.io/client-go/util/retry.OnError.func1(0x5a, 0xa0, 0x5a)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:51 +0x3c
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc000483ab8, 0xc000483aa0, 0xc000483b70)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:292 +0x51
k8s.io/client-go/util/retry.OnError(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0x153be18, 0xc000483b68, 0xc000483aa0, 0x4e599a)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:50 +0xa6
k8s.io/client-go/util/retry.RetryOnConflict(...)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:104
github.com/skupperproject/skupper/client.(*VanClient).ConnectorCreate(0xc0002c4000, 0x168e580, 0xc000044230, 0xc0004f4000, 0xc0004948a0, 0xa, 0xc000496620, 0x1b, 0x0, 0xc0002cbe60, ...)
        /go/src/app/client/connector_create.go:169 +0x12f
main.(*SiteController).connect(0xc00037a410, 0xc0004f4000, 0xc0004948a0, 0xa, 0xc0004948ab, 0x1b)
        /go/src/app/cmd/site-controller/controller.go:309 +0x1b6
main.(*SiteController).checkToken(0xc00037a410, 0xc0004948a0, 0x26, 0xc000483d68, 0x12012fc)
        /go/src/app/cmd/site-controller/controller.go:374 +0x272
main.(*SiteController).dispatchTrigger(0xc00037a410, 0xc0004948a0, 0x26, 0x1, 0x0, 0x0)
        /go/src/app/cmd/site-controller/controller.go:188 +0x121
main.(*SiteController).processNextTrigger(0xc00037a410, 0x0)
        /go/src/app/cmd/site-controller/controller.go:174 +0x10f
main.(*SiteController).run(0xc00037a410)
        /go/src/app/cmd/site-controller/controller.go:153 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000183490)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000183490, 0x3b9aca00, 0x0, 0x1, 0xc00009e660)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000183490, 0x3b9aca00, 0xc00009e660)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88 +0x4d
created by main.(*SiteController).Run
        /go/src/app/cmd/site-controller/controller.go:131 +0x423
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x11dd3bf]

goroutine 88 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x13223a0, 0x200fe10)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/skupperproject/skupper/client.(*VanClient).ConnectorCreate.func1(0x0, 0x0)
        /go/src/app/client/connector_create.go:196 +0x22f
k8s.io/client-go/util/retry.OnError.func1(0x5a, 0xa0, 0x5a)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:51 +0x3c
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc000483ab8, 0xc000483aa0, 0xc000483b70)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:292 +0x51
k8s.io/client-go/util/retry.OnError(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0x153be18, 0xc000483b68, 0xc000483aa0, 0x4e599a)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:50 +0xa6
k8s.io/client-go/util/retry.RetryOnConflict(...)
        /go/pkg/mod/k8s.io/client-go@v0.17.0/util/retry/util.go:104
github.com/skupperproject/skupper/client.(*VanClient).ConnectorCreate(0xc0002c4000, 0x168e580, 0xc000044230, 0xc0004f4000, 0xc0004948a0, 0xa, 0xc000496620, 0x1b, 0x0, 0xc0002cbe60, ...)
        /go/src/app/client/connector_create.go:169 +0x12f
main.(*SiteController).connect(0xc00037a410, 0xc0004f4000, 0xc0004948a0, 0xa, 0xc0004948ab, 0x1b)
        /go/src/app/cmd/site-controller/controller.go:309 +0x1b6
main.(*SiteController).checkToken(0xc00037a410, 0xc0004948a0, 0x26, 0xc000483d68, 0x12012fc)
        /go/src/app/cmd/site-controller/controller.go:374 +0x272
main.(*SiteController).dispatchTrigger(0xc00037a410, 0xc0004948a0, 0x26, 0x1, 0x0, 0x0)
        /go/src/app/cmd/site-controller/controller.go:188 +0x121
main.(*SiteController).processNextTrigger(0xc00037a410, 0x0)
        /go/src/app/cmd/site-controller/controller.go:174 +0x10f
main.(*SiteController).run(0xc00037a410)
        /go/src/app/cmd/site-controller/controller.go:153 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000183490)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000183490, 0x3b9aca00, 0x0, 0x1, 0xc00009e660)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000183490, 0x3b9aca00, 0xc00009e660)
        /go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88 +0x4d
created by main.(*SiteController).Run
        /go/src/app/cmd/site-controller/controller.go:131 +0x423
```

**Comment by grs** [comment]
I'm sorry. This was my fault again. I have a fix on master and on the 0.5 branch and will get a new release out soon.

**Comment by kungfuchicken** [comment]
np! thanks! :D

**Comment by grs** [comment]
Fixed in 0.5.2
